ext/ScreenShot/ScreenShot.h

```diff
@@ -5,27 +5,151 @@
 
 #include "../source/Irrlicht/COpenGLBuffer.h"
 #include "../source/Irrlicht/COpenGLExtensionHandler.h"
+#include "irr/asset/IImageAssetHandlerBase.h"
 
 namespace irr
 {
-namespace ext
-{
-namespace ScreenShot
-{
-	
-//! TODO: HANDLE UNPACK ALIGNMENT
-core::smart_refctd_ptr<video::IDriverFence> createScreenShot(video::IDriver* driver, video::IGPUImage* source, video::IGPUBuffer* destination, uint32_t sourceMipLevel=0u, size_t destOffset=0ull, bool implicitflush=true)
-{
-	// will change this, https://github.com/buildaworldnet/IrrlichtBAW/issues/148
-	if (isBlockCompressionFormat(source->getCreationParameters().format))
-		return nullptr;
+	namespace ext
+	{
+		namespace ScreenShot
+		{
+			/*
+				Creates useful FBO that may be used for instance 
+				to fetch default render target color data attachment 
+				after rendering scene.
 
-	auto extent = source->getMipSize(sourceMipLevel);
-	video::IGPUImage::SBufferCopy pRegions[1u] = { {destOffset,extent.x,extent.y,{static_cast<asset::IImage::E_ASPECT_FLAGS>(0u),sourceMipLevel,0u,1u},{0u,0u,0u},{extent.x,extent.y,extent.z}} };
-	driver->copyImageToBuffer(source,destination,1u,pRegions);
+				The usage is following:
 
-	return driver->placeFence(implicitflush);
-}
+				- create frame buffer using the function
+				- blit framebuffers, driver->blitRenderTargets(nullptr, frameBuffer, false, false);
+				Note that in the call above we don't want to copy depth and stencil buffer, but event though default
+				FBO contains depth buffer.
+				- pass frame buffer to performScreenShot(video::IFrameBuffer*)
+
+				Notes:
+
+				- color buffer is placed under video::EFAP_COLOR_ATTACHMENT0 attachment
+				- depth buffer is placed under video::EFAP_DEPTH_ATTACHMENT attachment
+			*/
+
+			irr::video::IFrameBuffer* createDefaultFBOForScreenshoting(core::smart_refctd_ptr<IrrlichtDevice> device)
+			{
+				auto driver = device->getVideoDriver();
```

Comment:
make the driver a parameter to this function + take a resolution parameter not a whole `device`

split the function in two, make default FBO + make default arrachments